Add dark theme

aelmahmoudy opened the pull request:
The problem with the last commit, is that when changing the theme, then clicking on a contact, Xabber crashes.

Actually adding the dark theme, made me discover an issue in the menu (probably related to https://github.com/redsolution/xabber-android/issues/372), is that the menu theme is light even if I use the dark theme.

aelmahmoudy commented:
Screenshot of the dark theme
![dark-theme](https://cloud.githubusercontent.com/assets/96682/6506907/045ad930-c357-11e4-80a8-95c6277959bb.png)

aelmahmoudy commented:
Despite the theme is dark, yet the menu is still white !
![white-menu](https://cloud.githubusercontent.com/assets/96682/6506923/2579bb4a-c357-11e4-8136-f77ed2ac0dff.png)

kr428 commented:
Is this available in any build already?  Just noticed that what seems the latest fdroid build is all bright ...

aelmahmoudy commented:
@kr428 no, it isn't available in a build

aelmahmoudy commented:
Hello, after rebasing on commit https://github.com/redsolution/xabber-android/commit/b1a74a27d718706f34b892603d1d7b2d8a7692fa, Xabber no longer crashes when the theme is changed !

aelmahmoudy commented:
Note: removed "Set the default them in AM.xml to DarkTheme" commit, as it is no longer needed.

aelmahmoudy commented:
Rebased on today's develop branch. Please note the following:
* I changed the chat_background images to grayscale+alpha (that saved about 100 kB of space per each image)
* There is an issue with the account select spinner with dark theme.

aelmahmoudy commented:
Drawer screenshot:

![drawer](https://cloud.githubusercontent.com/assets/96682/7633438/a1ba048a-fa5c-11e4-8f86-2230d35e884d.png)

aelmahmoudy commented:
Other screenshots for contacts list (still need to fix alpha? of the group backgrounds)
![contacts-compact](https://cloud.githubusercontent.com/assets/96682/7633876/f2d62a4e-fa5f-11e4-9e73-efd4550e443b.png)

![contacts](https://cloud.githubusercontent.com/assets/96682/7633878/f85be36e-fa5f-11e4-9426-4960c170ef0e.png)

moorchegue commented:
any progress on this?

aelmahmoudy commented:
@moorchegue You can try it. I am using it regularly.

moorchegue commented:
If it's working, shouldn't it be incorporated?

grigoryfedorov commented:
Dark theme finally implemented, based on this pull request (which, I think, based on previous versions of Xabber)
Thanks, @aelmahmoudy

moorchegue commented:
Terrific, can't wait to see the new build in fdroid! Thank you, guys!
